docs/styleguide/generic.rst

```diff
@@ -162,3 +162,43 @@ Comments
       do_something()
 
 * Avoid end of line comments.
+
+Git
+===
+
+Feature branch development
+--------------------------
+
+The code of our project is shared by all members of the team.
+
+We often have a feature branch that is driven by a single developer, but during
+the branch lifetime or the review process that branch may get contributions from
+other developers. For example typo-fixes or small GUI fixes.
+
+To reduce conflicts and confusion between colleagues, avoid using git commands
+that rewrite the history (rebase or commit with amend).
```

Comment:
You can only amend commits that are not yet pushed, right? Otherwise you are only amending your local commit and will end up with a conflict with HEAD.  I use `git commit --amend` quite often, would not recommend against it…

Reply:
Indeed, not-yet-pushed commits are not a problem. Will modify!

```suggestion
that rewrite the history of commits you have pushed (rebase or commit with amend).
```